src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableActorContext.cs

```diff
@@ -111,5 +123,55 @@ internal override void ThrowIfInvalidAccess()
                 throw new InvalidOperationException("No operation is being processed.");
             }
         }
+
+        internal override void SendActorMessage(OrchestrationInstance target, string eventName, object eventContent)
+        {
+            lock (this)
```

Comment:
The guidance I've seen recommends locking a `private readonly object` variable rather than `this` to avoid the possibility of some external code also locking the same object (and creating a hard-to-debug deadlock). You may want to do the same here as well.

Reply:
Right. I think it can simplify this a bit.